src/eureka/S1_detector_processing/bias_sub.py

```diff
@@ -0,0 +1,187 @@
+
+    # Replace NaN's in the superbias with zeros
+    bias_model.data[np.isnan(bias_model.data)] = 0.0
+
+    # Create 4D bias array for each integration/group and apply scale factor 
+    bias_data = np.ones((nint, ngroup, nrow, ncol)) * \
+        bias_model.data[np.newaxis, np.newaxis, :, :]
+    
+    try:
+        if meta.bias_correction == "mean":
+            if isinstance(meta.bias_group, int):
+                # Scale superbias frame using single value for all groups
+                jj = meta.bias_group
+                assert jj > 0, f"Group number should be >0, got: {jj}"
+                assert jj <= ngroup, f"Group number should be <={ngroup}" + \
+                    ", got: {jj}"
+                log.writelog('  Applying mean bias correction using ' +
+                             f'group {jj}.')
+                bias_model.data *= mean_scale_factor[jj-1]
+            elif meta.bias_group == "each":
+                # Scale superbias frame using means values for each group
+                log.writelog('  Applying mean bias correction to each group.')
+                bias_model.data *= mean_scale_factor[0]
+                bias_data *= mean_scale_factor[np.newaxis, :, np.newaxis, 
+                                               np.newaxis]
+            else:
+                raise ValueError('Incorrect meta.bias_group value: ' + 
+                                 f'{meta.bias_group}. Should be ' +
+                                 '[1, 2, ..., each].')
+        elif (meta.bias_correction == "group_level") or \
+             (meta.bias_correction == "smooth"):
+            # Apply correction for zeroframe
+            bias_model.data *= mean_scale_factor[0]
+            if isinstance(meta.bias_group, int):
+                # Scale superbias frame using values from one group
+                jj = meta.bias_group
+                assert jj > 0, f"Group number should be >0, got: {jj}"
+                assert jj <= ngroup, f"Group number should be <={ngroup}" + \
+                    ", got: {jj}"
+                log.writelog('  Applying group-level bias correction using ' +
+                             f'group {jj}.')
+                bias_data *= scale_factor[:, jj-1, np.newaxis, 
+                                          np.newaxis, np.newaxis]
+            elif meta.bias_group == "each":
+                # Scale superbias frame using values from each group
+                log.writelog('  Applying group-level bias correction to ' + 
+                             'each group.')
+                bias_data *= scale_factor[:, :, np.newaxis, np.newaxis]
+            else:
+                raise ValueError('Incorrect meta.bias_group value: ' + 
+                                 f'{meta.bias_group}. Should be ' +
+                                 '[1, 2, ..., each].')
+        else:
+            log.writelog('  No bias correction applied.')
+            bias_data = None
+    except Exception as error:
+        print(repr(error))
```

Comment:
Should this be logged instead of printed? If it is logged, it should be forced to be verbose